File: main/src/main/java/com/sedmelluq/lavaplayer/core/player/track/AudioTrackFactory.java
package com.sedmelluq.lavaplayer.core.player.track;

import com.sedmelluq.lavaplayer.core.player.configuration.AudioConfiguration;
import com.sedmelluq.lavaplayer.core.player.track.AudioTrackRequest;
import com.sedmelluq.lavaplayer.core.player.track.ExecutableAudioTrack;
import java.io.Closeable;

public interface AudioTrackFactory extends Closeable {
  ExecutableAudioTrack create(AudioTrackRequest request, AudioConfiguration configuration);
}

File: main/src/main/java/lavaplayer/track/AudioTrackFactory.kt
package lavaplayer.track

fun interface AudioTrackFactory {
    /**
     * Creates a new [AudioTrack] from the provided [AudioTrackInfo]
     *
     * @param info The info to create the new [AudioTrack] from.
     *
     * @return a new [AudioTrack]
     */
    fun create(info: AudioTrackInfo): AudioTrack
}
